### PR TITLE
refactor: use go modules

### PR DESCRIPTION
**Pull request by philoserf**
The build image was already go1.11. By using go modules the Dockerfile can be far simplier.

Signed-off-by: Mark Ayers <mark@philoserf.com>

---
This change is [Reviewable](https://reviewable.io/reviews/themagicalkarp/kube-janitor/7)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,34 @@
+module github.com/philoserf-contribs/kube-janitor
```
Comment (theMagicalKarp) [inline go.mod]: @philoserf Should this be `module github.com/theMagicalKarp/kube-janitor` to match the repo? Thanks for making a contribution btw!

Comment (philoserf) [reply go.mod]: Yes, it should. Standby...

**Comment by theMagicalKarp** [comment]
+10 looks good

Verified dependencies were correct by running `go mod init && go mod tidy` on master. I got the same results, with the exception of `github.com/pmezard/go-difflib`. I'm not worried about that repo though. I also ensured changes worked functionally in minikube via the helm install.

Overall pretty neat, I wasn't aware of go modules and it seems like the future of golang dependency management. Thanks @philoserf

**Comment by philoserf** [comment]
BTW: to fully update go modules config use: `go get -u&&go mod tidy`

**Pull request merged**